examples/tutorials/date_time_charts.py

```diff
@@ -0,0 +1,366 @@
+"""
+Plotting datetime charts
+========================
+
+Creating datetime charts is handled by :meth:`pygmt.Figure.basemap`.
+
+Plotting datetime data points is handled by :meth:`pygmt.Figure.basemap`.
+"""
+# sphinx_gallery_thumbnail_number = 0
+
+import datetime
+
+import numpy as np
+import pandas as pd
+import pygmt
+import xarray as xr
+
+########################################################################################
+# Datetime Input Types
+# --------------------
+#
+# PyGMT accepts a variety of datetime objects to plot data and create charts.
+# Aside from the built-in Python datetime object, PyGmt supports input using
+# ``pandas``, ``numpy``, ``xarray`` as well as properly formatted strings.
+# These data types can be used to plot specific points as well as get
+# passed into the ``region`` parameter to create a range of the data on an axis.
+#
+# The following examples will demonstrate how to create plots
+# using the differnt datetime objects
+
+###############################################################################
+# Using Python's ``Datetime``
+# ---------------------------
+#
+# In this example, Python's built-in ``datetime`` package is used
+# to create data points stored in list ``x``. The format of the
+# data is ``(Y, M, D)``. Additionally, dates are passed into the
+# ``region`` parameter in the format
+# ``(x_start, x_end, y_start, y_end)``,
+# where the date range is plotted on the x-axis.
+# An additonal notable parameter is `style`, where it's specified
+# that data points are to be plotted in an **X** shape with a size
+# of 0.3 centimeters.
+#
+
+x = [
+    datetime.date(2010, 6, 1),
+    datetime.date(2011, 6, 1),
+    datetime.date(2012, 6, 1),
+    datetime.date(2013, 6, 1),
+]
+y = [1, 2, 3, 5]
+
+fig = pygmt.Figure()
+fig.plot(
+    projection="X10c/5c",
+    region=[datetime.date(2010, 1, 1), datetime.date(2014, 12, 1), 0, 6],
+    frame=["WSen", "afg"],
+    x=x,
+    y=y,
+    style="x0.3c",
+    pen="1p",
+)
+
+fig.show()
+
+###############################################################################
+# In addition to specifying the date, ``datetime`` supports
+# the exact time of the data points. Using :meth:`datetime.datetime`
+# the ``region`` paramter as well as data points can be created
+# using this format ``(Y, M, D, H, M, S)`` where ``H`` is hours
+# using 24 hour format.
+#
+# Some notable differences to the previous example include
```

Comment:
This paragram could be merged into the last one, and I think we don't need to explain the ``frame`` parameter in this tutorial. How do you think?